sdks/python/apache_beam/dataframe/io.py

```diff
@@ -192,14 +207,133 @@ def expand(self, root):
                 self.reader,
                 self.args,
                 self.kwargs,
+                self.binary,
                 self.incremental,
-                self.splittable,
-                self.binary)))
+                self.splitter)))
     from apache_beam.dataframe import convert
     return convert.to_dataframe(
         pcoll, proxy=_prefix_range_index_with(':', sample[:0]))
 
 
+class _Splitter:
+  def empty_buffer(self):
+    raise NotImplementedError(self)
+
+  def read_header(self, handle):
+    raise NotImplementedError(self)
+
+  def read_to_record_boundary(self, buffered, handle):
+    raise NotImplementedError(self)
```

Comment:
Could you add docstrings and/or typehints for this class so its clear what implementations should do

Reply:
I actually tried to add typehints initially; they work poorly due to the fact that this works both on bytes and strings (but it's hard for mypy to know that it will be all one or all the other). But I added some comments at least.